marshmallow/fields.py

```diff
@@ -623,8 +623,11 @@ def _validated(self, value):
         if isinstance(value, uuid.UUID):
             return value
         try:
-            return uuid.UUID(value)
-        except (ValueError, AttributeError):
+            if isinstance(value, bytes) and len(value) == 16:
```

Comment:
Scratch my previous statement: I see now why you added the explicit length check--Python 2 `bytes` and `str` are the same type 😞 . So I just added exception handling for `TypeError` when an invalid input is passed to `uuid.UUID(value)` below.